### PR TITLE
Ease setup in Amazon Linux 2

### PR DESCRIPTION
**Pull request by adpaco-aws**
Adds a set of scripts to automate building and installing most dependencies in Amazon Linux 2 environments.

This will make it easier to provide ARM64 binaries for Linux (see #2805). The installation of dependencies isn't minimal but improves our situation w.r.t. AL2.

Testing done manually on remote AL2 ARM64 instances, after running this I'm able to run the regression successfully.

### Callouts
 * Hasn't been tested on AL2 x86.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: scripts/setup/al2/reinstall_cmake.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+set -eux
+
+CMAKE_VERSION="3.27.7"
+
+# Remove other versions of CMake
+sudo yum -y remove cmake
```
Comment (zhassan-aws) [inline scripts/setup/al2/reinstall_cmake.sh:10]: It's a bit unexpected for a setup script to delete an existing version of a program. Can you clarify why this is needed?

**Comment by celinval** [comment]
Can you install rustup as well in your scripts?

**Comment by adpaco-aws** [comment]
> It's a bit unexpected for a setup script to delete an existing version of a program.

We now avoid building `cmake` from source by installing `cmake3` and using that to build CBMC.

> Can you install rustup as well in your scripts?

Sure, just added that too.

File: scripts/setup/al2/reinstall_cmake.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+wget https://github.com/Kitware/CMake/releases/download/v"${CMAKE_VERSION}"/cmake-"${CMAKE_VERSION}".tar.gz
+tar -xzvf cmake-"${CMAKE_VERSION}".tar.gz
+cd cmake-"${CMAKE_VERSION}"
+
+./bootstrap
+make -j$(nproc)
+sudo make install
+
+popd
```
[the remark this answers is not in the document]
Comment (adpaco-aws) [reply scripts/setup/al2/reinstall_cmake.sh]: This script isn't added anymore.

File: scripts/setup/al2/reinstall_cmake.sh (hunk 1 above)
Comment (adpaco-aws) [reply scripts/setup/al2/reinstall_cmake.sh]: Tested that this works, resolving.

File: scripts/setup/al2/install_deps.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+set -eu
+
+# Dependencies.
+# Note: CMake 3.8 or higher is required to build CBMC, but those versions are
+# only available in AWS AMIs through `cmake3`. So we install `cmake3` and use it
+# to build CBMC.
+DEPS=(
+  cmake
```
Comment (zhassan-aws) [inline scripts/setup/al2/install_deps.sh:12]: Is `cmake` needed?

**Review by zhassan-aws — APPROVED** [review]

**Pull request merged**